### ISSUE TITLE
[Offset computation] trouble understanding the code

### ISSUE THREAD

**Issue by joihn** [ultralytics/yolov5#6254]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Thanks for this repo !

I'm having trouble understanding how the offset computation works, in `utils/loss.py:build_targets()`
```
  # Offsets
  gxy = t[:, 2:4]  # grid xy
  gxi = gain[[2, 3]] - gxy  # inverse
  j, k = ((gxy % 1 < g) & (gxy > 1)).T
  l, m = ((gxi % 1 < g) & (gxi > 1)).T
  j = torch.stack((torch.ones_like(j), j, k, l, m))
  t = t.repeat((5, 1, 1))[j]
  offsets = (torch.zeros_like(gxy)[None] + off[:, None])[j]
```

What does `j,k,l,m` represent ?

**Comment by glenn-jocher** [comment]
@joihn these variables are indices for labels intended for matching with an anchor
